KerbalStuff/thumbnail.py

```diff
@@ -1,12 +1,48 @@
 import os.path
-from typing import Tuple
 
 from PIL import Image
 
+from KerbalStuff.config import _cfg, _cfgi, site_logger
 
-def create(imagePath: str, thumbnailPath: str, thumbnailSize: Tuple[int, int]) -> None:
-    if not os.path.isfile(imagePath):
+
+def create(background_path: str, thumbnail_path: str) -> None:
+    if not os.path.isfile(background_path):
         return
-    im = Image.open(imagePath)
-    im.thumbnail(thumbnailSize, Image.ANTIALIAS)
-    im.save(thumbnailPath, 'jpeg', quality=50, optimize=True)
+
+    size_str = _cfg('thumbnail_size')
+    if not size_str:
+        size_str = "320x320"
+    size_str_tuple = size_str.split('x')
+    size = (int(size_str_tuple[0]), int(size_str_tuple[1]))
+
+    quality = _cfgi('thumbnail_quality')
+    if not quality or quality > 100:
+        quality = 80
+
+    im = Image.open(background_path)
+    if im.mode in ("RGBA", "P"):
+        im = im.convert("RGB")
+    im.thumbnail(size, Image.ANTIALIAS)
```

Comment:
Is `ANTIALIAS` supported for this parameter? The documentation does not list it:

https://pillow.readthedocs.io/en/5.1.x/reference/Image.html#PIL.Image.Image.thumbnail

> - resample – Optional resampling filter. This can be one of PIL.Image.NEAREST, PIL.Image.BILINEAR, PIL.Image.BICUBIC, or PIL.Image.LANCZOS. If omitted, it defaults to PIL.Image.BICUBIC. (was PIL.Image.NEAREST prior to version 2.5.0)

Maybe the artifacts that @V1TA5 is always complaining about are related to an invalid parameter being passed here?

... older docs say it was renamed:

https://pillow.readthedocs.io/en/3.0.x/releasenotes/2.7.0.html#antialias-renamed-to-lanczos

> A new PIL.Image.LANCZOS constant was added instead of ANTIALIAS.
> 
> When ANTIALIAS was initially added, it was the only high-quality filter based on convolutions. It’s name was supposed to reflect this. Starting from Pillow 2.7.0 all resize method are based on convolutions. All of them are antialias from now on. And the real name of the ANTIALIAS filter is Lanczos filter.
> 
> The ANTIALIAS constant is left for backward compatibility and is an alias for LANCZOS.

Reply:
I'll try out using `Image.LANCZOS` instead and check whether it does what we want (it probably does).